spec/abci/abci++_app_requirements.md

```diff
@@ -1015,6 +1015,39 @@ from the genesis file and light client RPC servers. It also calls `Info` to veri
   current height's block header
 
 Once the state machine has been restored and CometBFT has gathered this additional
-information, it transitions to block sync (if enabled) to fetch any remaining blocks up the chain
-head, and then transitions to regular consensus operation. At this point the node operates like
-any other node, apart from having a truncated block history at the height of the restored snapshot.
+information, it transitions to consensus. As of ABCI 2.0, CometBFT ensures the neccessary conditions
+to switch are met [RFC-100](./../../docs/rfc/rfc-100-abci-vote-extension-propag.md#base-implementation-persist-and-propagate-extended-commit-history).
+From the application's point of view, these operations are transparent, unless the application has just upgraded to ABCI 2.0. 
+In that case, the application needs to be properly configured and aware of certain constraints in terms of when
+to provide vote extensions. More details can be found in the section below. 
+
+Once a node switches to consensus,it operates like any other node, apart from having a truncated block history at the height of the restored snapshot.
+
+## Application configuration required to switch to ABCI 2.0
+
+Introducing vote extensions requires changes to the configuration of the application.
+
+First of all, switching to a version of CometBFT with vote extensions, requires a coordinated upgrade. 
+For a detailed description on the upgrade path, please refer to the corresponding 
+[section](./../../docs/rfc/rfc-100-abci-vote-extension-propag.md#upgrade-path) in RFC-100.
+
+The configuration file now contains a [**new consensus parameter**](./abci%2B%2B_app_requirements.md#abciparamsvoteextensionsenableheight): `VoteExtensionsEnableHeight`. 
+This parameter represents the height after which vote extensions are 
+required for consensus to proceed, with 0 being the default value (no vote extensions).
```

Comment:
@sergio-mena , maybe I have not a clear understanding, but let's say a chain is just starting and wants to have vote extensions, but the default value here is 0. Should we not say something like - we have to set this to at least 1? Because someone starting from scratch with a version of the software having extensions enabled, might assume they are enabled by default?

Reply:
Good point.
First, I realize we're not mentioning that the ConsensusParam can be set at genesis. So, typically, for chains starting from scratch and having vote extensions from their app's first version, they just need to set `VoteExtensionsEnableHeight` to 1 at genesis and forget about the upgrade path. We need to say this somewhere in the spec.
Second, to your question. We (our team) discussed about the default value at genesis last spring in some PR (I'll try to pull it up). I remember thinking default value at genesis should be 1, but then people convincing me that 0 is better. Now, when I think of it, it makes total sense to me that the default value is zero, for compatibility with existing genesis docs.
Let me know if you're still unclear in some aspect.

Reply:
> they just need to set VoteExtensionsEnableHeight to 1 
actually, to be precise, they need to set it to their `InitialHeight` parameter

Reply:
Yes, I know and agree it should be 0 by default. But as you said, I think we should state it that, if they do want vote extensions they have to change it, they will not magically be turned on. Of course, it can be deducted from the spec, but I think it's better to be clear on this as people might just skim through some parts of the spec.

Reply:
> I'll try to pull it up

[Here](https://github.com/tendermint/tendermint/issues/8453#issuecomment-1116135296) it is (took some time :-) )

Reply:
Agreed.

Just make sure it's clear that there are two way of enabling vote extensions:
* Via app's logic at a given height, by changing the ConsensusParam
* Via setting it to `InitialHeight` in the genesis. This way, the app does not need to do anything to have extensions enabled from the very beginning.
BTW, this distinction also applies to other parts of CometBFT's state, such as validator updates.